scripts/verify.py

```diff
@@ -0,0 +1,184 @@
+from manticore import Manticore
+from manticore.platforms import linux_syscalls
+
+import logging
+
+from sys import argv, exit
+import struct
+import qemu
+import gdb
+
+logger = logging.getLogger('TRACE')
+
+## We need to keep some complex objects in between hook invocations so we keep them
+## as globals. Tracing is inherently a single-threaded process, so using a 
+## manticore context would be heavier than needed.
+stack_top = 0xc0000000
+stack_size = 0x20000
+icount = 0
+initialized = False
+last_instruction = None
+syscall = 0
+in_helper = False
+
+def dump_gdb(cpu, addr, count):
+    for offset in range(addr, addr+count, 4):
+        val = int(gdb.getM(offset)  & 0xffffffff)
+        val2 = int(cpu.read_int(offset))
+        print '{:x}: g{:08x} m{:08x}'.format(offset, val, val2)
+
+def cmp_regs(cpu, should_print=False):
+    '''
+    Compare registers from a remote gdb session to current mcore.
+
+    :param manticore.core.cpu Cpu: Current cpu
+    :param bool should_print: Whether to print values to stdout
+    :return: Whether or not any differences were detected
+    :rtype: bool
+    '''
+    differing = False
+    gdb_regs = gdb.getCanonicalRegisters()
+    for name in sorted(gdb_regs):
+        vg = gdb_regs[name]
+        if name.endswith('psr'):
+            name = 'apsr'
+        v = cpu.read_register(name.upper())
+        if should_print:
+            print '{} gdb:{:x} mcore:{:x}'.format(name, vg, v)
+        if vg != v:
+            if should_print:
+                print '^^ unequal'
+            differing = True
+    return differing
+
+def on_after(state, last_instruction):
```

Comment:
`on_after` doesn't just verify instruction, it's supposed to be everything we need to handle after an instruction is executed, with the idea to move it to a hook when we get post-exec hooks.